tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/EmailRecoveryActionSetMethod.scala

```diff
@@ -157,13 +165,19 @@ class EmailRecoveryActionSetCreatePerformer @Inject()(val taskManager: TaskManag
     } yield parsedRequest
 
   private def submitTask(clientId: EmailRecoveryActionCreationId, userToRestore: Username, creationRequest: EmailRecoveryActionCreationRequest): SMono[CreationResult] = {
-    val fifteenDaysAgo = ZonedDateTime.now().minusDays(15)
-    val horizonCriterion = CriterionFactory.deletionDate().afterOrEquals(fifteenDaysAgo)
-
-    val modifiedQuery = Query.and(ImmutableList.builder()
-      .addAll(creationRequest.asQuery(configuration.maxEmailRecoveryPerRequest).getCriteria)
-      .add(horizonCriterion)
-      .build())
+    val horizonCriterion = LazyList(configuration)
+      .map(_.userHorizon.toDays)
+      .map(ZonedDateTime.now().minusDays)
+      .map(CriterionFactory.deletionDate().afterOrEquals)
+      .head
```

Comment:
Why going through a list?

```suggestion
val days: Int = configuration.restorationHorizon.toDays
val horizon: ZoneDateTime = ZonedDateTime.now().minusDays(days)
val horizonCriterion = CriterionFactory.deletionDate().afterOrEquals(horizon)
```

?

Also why force a day granularity? What if I say: horizon is 1h ?


```suggestion
val horizon: ZoneDateTime = ZonedDateTime.now().minus(configuration.restorationHorizon)
val horizonCriterion = CriterionFactory.deletionDate().afterOrEquals(horizon)
```